src/Ryujinx.Memory/MemoryManagementUnix.cs

```diff
@@ -186,17 +186,21 @@ public unsafe static IntPtr CreateSharedMemory(ulong size, bool reserve)
                 throw new OutOfMemoryException();
             }
 
-            return (IntPtr)fd;
+            return fd;
         }
 
         public static void DestroySharedMemory(IntPtr handle)
         {
+#pragma warning disable CA2020 // Prevent behavioral change caused by built-in operators of IntPtr/UIntPtr
             close((int)handle);
+#pragma warning restore CA2020
         }
 
         public static IntPtr MapSharedMemory(IntPtr handle, ulong size)
         {
+#pragma warning disable CA2020 // Prevent behavioral change caused by built-in operators of IntPtr/UIntPtr
             return mmap(IntPtr.Zero, size, MmapProts.PROT_READ | MmapProts.PROT_WRITE, MmapFlags.MAP_SHARED, (int)handle, 0);
```

Comment:
```suggestion
            return mmap(IntPtr.Zero, size, MmapProts.PROT_READ | MmapProts.PROT_WRITE, MmapFlags.MAP_SHARED, handle.ToInt32(), 0);
```
And remove the pragma